sherpa-onnx/jni/jni_java.cc

```diff
@@ -0,0 +1,414 @@
+// sherpa-onnx/jni/jni_java.cc
+// It is based on jni.cc, but with some changes
+// Copyright   2022-2023  by zhaoming
+// this will be used for the java environment
+// com.k2fsa.sherpaonnx.rcglib.OnlineRecognizer
+// It support for parallel stream decoding and each
+// stream will assign a uuid id.
+//
+
+#include "jni.h"  // NOLINT
+
+#include <strstream>
+#include <utility>
+
+#include <fstream>
+#include <map>
+#include <random>
+#include <sstream>
+#include "sherpa-onnx/csrc/macros.h"
+#include "sherpa-onnx/csrc/online-recognizer.h"
+#include "sherpa-onnx/csrc/onnx-utils.h"
+
+#define SHERPA_ONNX_EXTERN_C extern "C"
+#include <mutex>  // NOLINT
+namespace sherpa_onnx {
+
+static std::random_device rd;
+static std::mt19937 gen(rd());
+static std::uniform_int_distribution<> dis(0, 15);
+static std::uniform_int_distribution<> dis2(8, 11);
+
+std::string generate_uuid() {
```

Comment:
Could you document that the code is from
https://stackoverflow.com/questions/24365331/how-can-i-generate-uuid-in-c-without-using-boost-library
?